packages/databricks-vscode/src/language/notebooks/NotebookInitScriptManager.ts

```diff
@@ -76,6 +178,216 @@ export class NotebookInitScriptManager implements Disposable {
+            if (!executable) {
+                window
+                    .showErrorMessage(
+                        "Cannot verify databricks notebook init script. No python interpretter selected.",
+                        "Select Python Interpretter"
+                    )
+                    .then((choice) => {
+                        if (choice === "Select Python Interpretter") {
+                            this.pythonExtension.selectPythonInterpreter();
+                        }
+                    });
+                return;
+            }
+
+            if (
+                !(await this.pythonExtension.findPackageInEnvironment(
+                    "ipython"
+                ))
+            ) {
+                window
+                    .showErrorMessage(
+                        `Cannot verify databricks notebook init script. IPython is not installed in the current environment: ${
+                            (await this.pythonExtension.pythonEnvironment)
+                                ?.environment?.name
+                        }`,
+                        "Install and try again",
+                        "Change environment"
+                    )
+                    .then((choice) => {
+                        switch (choice) {
+                            case "Install and try again":
+                                this.pythonExtension
+                                    .installPackageInEnvironment("ipython")
+                                    .then(() => {
+                                        this.verifyInitScript();
+                                    });
+                                return;
+                            case "Change environment":
+                                this.pythonExtension.selectPythonInterpreter();
+                                return;
+                        }
+                    });
+            }
+
+            await this.updateInitScript();
+
+            this.initScriptSuccessfullyVerified = fromCommand
+                ? await window.withProgress(
+                      {location: ProgressLocation.Notification},
+                      async (progress) => {
+                          progress.report({
+                              message:
+                                  "Verifying databricks notebook init scripts...",
+                          });
+                          const success =
+                              await this.executeAndVerifyInitScripts(
+                                  executable
+                              );
+                          if (success) {
+                              window.showInformationMessage(
+                                  "Successfully verified Databricks notebook init scripts"
+                              );
+                          }
+                          return success;
+                      }
+                  )
+                : await this.executeAndVerifyInitScripts(executable);
```

Comment:
Now withProgress is called only when we call `fromCommand` otherwise we remain silent (until error).